src/main/java/com/faforever/client/legacy/GameStatus.java

```diff
@@ -0,0 +1,20 @@
+package com.faforever.client.legacy;
+
+import com.faforever.client.legacy.domain.GameState;
+
+public enum GameStatus {
+  NONE, HOST, LOBBY, PLAYING;
+
+  //FIXME cannot resolve gamestatus from gamestate
+  public static GameStatus getFromGameState(GameState gameState) {
+    switch (gameState.getString()) {
+      case "playing":
+        return GameStatus.PLAYING;
+      case "open":
+        return GameStatus.LOBBY;
+      //unknown and closed
+      default:
+        return GameStatus.NONE;
+    }
+  }
+}
```

Comment:
Have a look how other enumerations resolve from string to value; this code needs to be adjusted whenever an value is added which is unnecessary and error prone. In fact it's already bugged; you haven't mapped`HOST` so it resolves to `NONE`.